README
NAME

    Dist::Zilla::Plugin::NoSmartCommentsTests - Make sure no
    Smart::Comments escape into the wild

VERSION

    This document describes version 0.008 of
    Dist::Zilla::Plugin::NoSmartCommentsTests - released August 07, 2015 as
    part of Dist-Zilla-Plugin-NoSmartCommentsTests.

SYNOPSIS

        ; In C<dist.ini>:
        [Test::NoSmartComments]

DESCRIPTION

    This is an extension of Dist::Zilla::Plugin::InlineFiles, providing the
    following file:

      xt/release/no-smart-comments.t - test to ensure no Smart::Comments

NOTE

    The name of this plugin has turned out to be somewhat misleading, I'm
    afraid: we don't actually test for the _existance_ of smart comments,
    rather we ensure that Smart::Comment is not used by any file checked.

BUGS

    All complex software has bugs lurking in it, and this module is no
    exception.

    Please report any bugs to
    "bug-dist-zilla-plugin-nosmartcommentstests@rt.cpan.org", or through
    the web interface at <http://rt.cpan.org>.

    Patches and pull requests through GitHub are most welcome; our page and
    repo (same URI):

        https://github.com/RsrchBoy/dist-zilla-plugin-nosmartcommentstests

SEE ALSO

    Please see those modules/websites for more information related to this
    module.

      * Smart::Comments

      * Test::NoSmartComments

BUGS

    Please report any bugs or feature requests on the bugtracker website
    https://github.com/RsrchBoy/dist-zilla-plugin-nosmartcommentstests/issues

    When submitting a bug or request, please include a test-file or a patch
    to an existing test-file that illustrates the bug or desired feature.

AUTHOR

    Chris Weyl <cweyl@alumni.drew.edu>

 I'm a material boy in a material world

    Please note I do not expect to be gittip'ed or flattr'ed for this work,
    rather it is simply a very pleasant surprise. I largely create and
    release works like this because I need them or I find it enjoyable;
    however, don't let that stop you if you feel like it ;)

    Flattr
    <https://flattr.com/submit/auto?user_id=RsrchBoy&url=https%3A%2F%2Fgithub.com%2FRsrchBoy%2Fdist-zilla-plugin-nosmartcommentstests&title=RsrchBoy's%20CPAN%20Dist-Zilla-Plugin-NoSmartCommentsTests&tags=%22RsrchBoy's%20Dist-Zilla-Plugin-NoSmartCommentsTests%20in%20the%20CPAN%22>,
    Gratipay <https://gratipay.com/RsrchBoy/>, or indulge my Amazon
    Wishlist <http://bit.ly/rsrchboys-wishlist>... If and *only* if you so
    desire.

COPYRIGHT AND LICENSE

    This software is Copyright (c) 2011 by Chris Weyl.

    This is free software, licensed under:

      The GNU Lesser General Public License, Version 2.1, February 1999



Changes
Changes for Dist::Zilla::Plugin::NoSmartCommentsTests

{{$NEXT}}

0.009     2016-10-09 19:05:49 CDT-0500
	* Add Test::NoSmartComments as a develop requires on a generated
	  distribution.

0.008-TRIAL  2015-08-07 18:32:43 PDT-0700
	* Switch over to using FileFinders rather than Test::NoSmartComments' own
	  file finding functionality.
	* Add a shim Test::NoSmartComments plugin

0.007     2014-12-02 19:51:38 PST-0800
    * check t/*.t, too.

0.006     2011-04-24 16:10:00 America/Los_Angeles
    * Update POD

0.005     2011-03-27 07:56:00 America/Los_Angeles
    * Add done_testing() to the end of our generated .t

0.004     2011-03-17 21:14:33 America/Los_Angeles
    * Switch to Test::NoSmartComments

0.003
    * AutoPrereqs wasn't picking up Module::ScanDeps -- so require it
      manually

0.002     2011-03-13 16:16:29 America/Los_Angeles
    * Metadata update

0.001     2011-03-13 16:09:56 America/Los_Angeles
    * Initial release



t/test-dists/fail/lib/DZ/Fail.pm
package DZ::Fail;
use strict;
use warnings;

 # ABSTRACT: turns baubles into trinkets

use Smart::Comments;

1;


t/test-dists/pass/lib/DZ/Pass.pm
package DZ::Pass;
use strict;
use warnings;

 # ABSTRACT: turns baubles into trinkets

1;


Makefile.PL
#
# This file is part of Dist-Zilla-Plugin-NoSmartCommentsTests
#
# This software is Copyright (c) 2011 by Chris Weyl.
#
# This is free software, licensed under:
#
#   The GNU Lesser General Public License, Version 2.1, February 1999
#
# This file was automatically generated by Dist::Zilla::Plugin::MakeMaker v5.037.
use strict;
use warnings;

use 5.006;

use ExtUtils::MakeMaker;

my %WriteMakefileArgs = (
  "ABSTRACT" => "Make sure no Smart::Comments escape into the wild",
  "AUTHOR" => "Chris Weyl <cweyl\@alumni.drew.edu>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0
  },
  "DISTNAME" => "Dist-Zilla-Plugin-NoSmartCommentsTests",
  "EXE_FILES" => [],
  "LICENSE" => "lgpl",
  "MIN_PERL_VERSION" => "5.006",
  "NAME" => "Dist::Zilla::Plugin::NoSmartCommentsTests",
  "PREREQ_PM" => {
    "Dist::Zilla::Plugin::InlineFiles" => 0,
    "Dist::Zilla::Role::FileFinderUser" => 0,
    "Dist::Zilla::Role::TextTemplate" => 0,
    "Moose" => 0,
    "MooseX::AttributeShortcuts" => 0,
    "Test::NoSmartComments" => "0.005",
    "autobox::Core" => 0,
    "namespace::autoclean" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "IO::Handle" => 0,
    "IPC::Open3" => 0,
    "Test::CheckDeps" => "0.010",
    "Test::More" => "0.94",
    "strict" => 0,
    "warnings" => 0
  },
  "VERSION" => "0.008",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Dist::Zilla::Plugin::InlineFiles" => 0,
  "Dist::Zilla::Role::FileFinderUser" => 0,
  "Dist::Zilla::Role::TextTemplate" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "IO::Handle" => 0,
  "IPC::Open3" => 0,
  "Moose" => 0,
  "MooseX::AttributeShortcuts" => 0,
  "Test::CheckDeps" => "0.010",
  "Test::More" => "0.94",
  "Test::NoSmartComments" => "0.005",
  "autobox::Core" => 0,
  "namespace::autoclean" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);
